shared/middleware/tables_api.rb

```diff
@@ -122,7 +122,7 @@ class TablesApi < Sinatra::Base
               "contains a name for each column:<br><br>#{columns.join(',')}"
         halt 400, {}, msg unless column
       end
-      CSV.foreach(tempfile, headers: true) do |row|
+      CSV.foreach(tempfile, headers: true, skip_blanks: true) do |row|
```

Comment:
Yeah the implicit hash thing is kind of confusing. I think you would only use braces if there were two consecutive hash arguments.